charts/m4d/templates/egeria-connector/egeria-connector-deployment.yaml

```diff
@@ -0,0 +1,63 @@
+{{- $autoFlag := and .Values.manager.enabled .Values.coordinator.enabled (eq .Values.coordinator.catalog "egeria") }}
+{{- if include "m4d.isEnabled" (tuple .Values.egeriaConnector.enabled $autoFlag) }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: egeria-connector
+  labels:
+    app.kubernetes.io/component: egeria-connector
+    {{- include "m4d.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.egeriaConnector.autoscaling.enabled }}
+  replicas: {{ .Values.egeriaConnector.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: egeria-connector
+      {{- include "m4d.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.egeriaConnector.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        m4d.ibm.com/componentType: connector
+        app.kubernetes.io/component: egeria-connector
+        {{- include "m4d.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.global.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ .Values.egeriaConnector.serviceAccount.name }}
+      securityContext:
+        {{- toYaml .Values.egeriaConnector.podSecurityContext | nindent 8 }}
+      containers:
+        - name: egeria-connector
+          securityContext:
+            {{- toYaml .Values.egeriaConnector.securityContext | nindent 12 }}
+          image: {{ include "m4d.image" ( tuple $ .Values.egeriaConnector ) }}
+          imagePullPolicy: {{ .Values.egeriaConnector.imagePullPolicy | default .Values.global.imagePullPolicy }}
+          ports:
```

Comment:
Are there some default liveness or readiness probes for egeria?

Reply:
This is the connector, not egeria itself. But you're right we can probably add it. I'll check if this can be done easily for GRPC like it is for HTTP.

Reply:
OK it's not that easy, the examples I found require a grpc_health_probe binary.